aten/src/ATen/native/quantized/cpu/qconv.cpp

```diff
@@ -244,6 +265,155 @@ class QConv2dInt8 final : public c10::OperatorKernel {
+
+    PackedConvWeightsQnnp& pack_ptr =
+        cpp_custom_type_hack::cast<PackedConvWeightsQnnp>(packed_weight);
+    auto packB = pack_ptr.w.get();
+    auto& kernel = pack_ptr.kernel;
+    auto kernel_zp = pack_ptr.w_zp;
+    auto kernel_scale = pack_ptr.w_scale;
+
+    const uint32_t kernel_h = kernel[0];
+    const uint32_t kernel_w = kernel[1];
+    const auto out_ch = packB->getOutputChannels();
+    // inputs are in NHWC format
+    Tensor input_contig = act.contiguous();
+    int N = input_contig.size(0);
+    int H = input_contig.size(1);
+    int W = input_contig.size(2);
+    int in_ch = input_contig.size(3);
+    int K = out_ch; // output channels
+
+    uint32_t stride_h = stride[0];
+    uint32_t stride_w = stride[1];
+    uint32_t pad_t = padding[0];
+    uint32_t pad_l = padding[1];
+    uint32_t dilation_h = dilation[0];
+    uint32_t dilation_w = dilation[1];
+
+    auto output_min = ReluFused
+        ? activationLimits(output_scale, output_zero_point, Activation::RELU)
+              .first
+        : std::numeric_limits<uint8_t>::min();
+    auto output_max = ReluFused
+        ? activationLimits(output_scale, output_zero_point, Activation::RELU)
+              .second
+        : std::numeric_limits<uint8_t>::max();
+    qnnpack::conv_param_t conv_p(
+        {kernel_w, kernel_h},
+        {stride_w, stride_h},
+        {dilation_w, dilation_h},
+        {pad_t, pad_l, pad_t, pad_l},
+        groups,
+        in_ch,
+        out_ch,
+        kernel_zp,
+        kernel_scale,
+        output_min,
+        output_max);
+
+    auto outShape =
+        convOutputShape(N, H, W, K, kernel, stride, padding, dilation);
+    TORCH_CHECK(
+        std::all_of(
+            outShape.begin(), outShape.end(), [](int64_t i) { return i > 0; }),
+        "quanitzed::conv2d (qnnpack): each dimension of output tensor should be greater "
+        "than 0")
+    TORCH_CHECK(
+        (outShape[3] == out_ch),
+        "quanitzed::conv2d (qnnpack): Number of filters must be equal to number of "
+        "output channels")
+
+    // Allocate output Tensor and a buffer for QNNPACK to use
+    Tensor output = at::_empty_affine_quantized(
+        outShape,
+        at::device(kCPU).dtype(kQUInt8),
+        output_scale,
+        output_zero_point);
+
+    const pytorch_qnnp_status runStatus = qnnpack::qnnpackConv(
+        conv_p,
+        packB->getPackedWeights(),
+        N,
+        H,
+        W,
+        input_contig.q_scale(),
+        input_contig.q_zero_point(),
+        (uint8_t*)input_contig.data_ptr<c10::quint8>(),
+        output.q_scale(),
+        output.q_zero_point(),
+        (uint8_t*)output.data_ptr<c10::quint8>(),
+        nullptr);
+
+    TORCH_INTERNAL_ASSERT(
+        runStatus == pytorch_qnnp_status_success,
+        "failed to run quanitzed::conv2d (qnnpack) operator");
+
+    return output;
+  }
+#endif
+#if defined(USE_FBGEMM) || defined(USE_PYTORCH_QNNPACK)
```

Comment:
I think you don't need this #ifdef. - you can just have one function and it'd throw if neither of them is available

If it's about warnings that arguments are not used - I think there's a way to disable the warnings.

Reply:
Sounds good. Removed the ifdef.